### ISSUE TITLE
Height is not being set properly

### ISSUE THREAD

**Issue by sagarnayak** [djyde/cusdis#283]
I have added the required partials to my layouts/_default/partials

but the height is not being set properly.

```
{{- /* Comments with cusdis.com */ -}}

<h2>Discussion:</h2>
<div id="cusdis_thread" data-host="https://auth.blog.hardcode.online" data-app-id="818a5966-bc74-4"
    data-page-id="{{ .File.UniqueID }}" data-page-url="{{ .Permalink }}" data-page-title="{{ .Title }}">
</div>
<script defer src="https://auth.blog.hardcode.online/js/widget/lang/fr.js"></script>
<script async defer src="https://auth.blog.hardcode.online/js/cusdis.es.js"></script>

{{- /* Comments area end */ -}}
```

and in single HTML
```
    <footer class="post-footer">
        {{/* Comments */}}
        {{- partial "comments.html" . }}
        <h4>Commentsss:</h4>
        {{- $tags := .Language.Params.Taxonomies.tag | default "tags" }}
        <ul class="post-tags">
            {{- range ($.GetTerms $tags) }}
            <li><a href="{{ .Permalink }}">{{ .LinkTitle }}</a></li>
            {{- end }}
        </ul>
        {{- if (.Param "ShowPostNavLinks") }}
        {{- partial "post_nav_links.html" . }}
        {{- end }}
        {{- if (and site.Params.ShowShareButtons (ne .Params.disableShare true)) }}
        {{- partial "share_icons.html" . -}}
        {{- end }}
    </footer>

    {{- if (.Param "comments") }}
    {{- partial "comments.html" . }}
    {{- end }}
</article>

{{- end }}{{/* end main */}}
```

what might be the reason? do i have to add some CSS here ?

this is what i see in my blog

![image](https://github.com/user-attachments/assets/b486528a-8209-4866-a11f-c5d9c916f526)

this should take the proper required height. instead this is taking a fixed height.

**Comment by Fluxwang** [comment]
Hi,  maybe you can take a look at this issue, [Issues#233](https://github.com/djyde/cusdis/issues/233#issuecomment-1877440099). I encountered the same problem, and it seems to be related to the iframe. When I switched my Docker image 'djyde/cusdis' to version 1.3.2, everything went back to normal.

**Comment by carl0schen** [comment]
I'm having the same problem with my comment system hosted at cusdis.com.

**Comment by seburbandev** [comment]
I was having the same problem where the height property wasn't set and I was getting a small window with a scroll bar. 

I am using the free tier, hosted at cusdis.com and using the embed code provided.

I've come up with a workaround by writing a short script to resize the iframe on page load. 

If you're using the embed code provided like I am, which looks like this

```
<div id="cusdis_id_here"
 data-host="https://cusdis.com"
 data-page-id="{{ PAGE_ID }}"
 data-page-url="{{ PAGE_URL }}"
 data-page-title="{{ PAGE_TITLE }}">
</div>
```

Then here is the script you need to include on your page to resize the iframe on each page reload.

```
window.addEventListener('load', function () {
            setTimeout(() => {
                let scrollHeight = document.querySelector("#cusdis_id_here iframe").contentWindow.document.body.scrollHeight;
                document.querySelector("#cusdis_id_here iframe").style.height = scrollHeight + "px";
            }, 3000);
        });
```


I had to add the timeout because the iframe seems to take awhile to load so we want to make sure the element we're targeting actually exists in dom.

Adjust the timeout as you like, but considering comments are usually at the bottom of the page (and users usually don't scroll to the bottom straight away), it's safer to give the iframe more time to load than adjust its height prematurely.

Your mileage may vary.

**Comment by rnkdev** [comment]
I'm also facing the same problem with comments hosted on cusdis.com, any thoughts @djyde ?

@seburbandev Appreciate your work around, I'll try it on my site. But I think this should be fixed on the source because timeout feature is not reliable depending on network load etc.

**Comment by seburbandev** [comment]
@rnkdev I completely agree, this should be fixed on source, and my approach is hacky and by no means a proper solution so use with caution.

That said, it seems that this issue had been raised a couple of times a while ago and still no response. In my opinion, the scroll makes it unusable, which is a real shame, so in my case it was either this or an alternative. I'm going with the former for now.

**Comment by kitbur** [comment]
I'm having the same issue: free tier, not self-hosted.

**Comment by schiessle** [comment]
I struggle with the same problem, and even Cusdis has the issue on their own site, e.g. https://cusdis.com/doc#/faq

**Comment by 4627488** [comment]
> I was having the same problem where the height property wasn't set and I was getting a small window with a scroll bar.
> 
> I am using the free tier, hosted at cusdis.com and using the embed code provided.
> 
> I've come up with a workaround by writing a short script to resize the iframe on page load.
> 
> If you're using the embed code provided like I am, which looks like this
> 
> ```
> <div id="cusdis_id_here"
>  data-host="https://cusdis.com"
>  data-page-id="{{ PAGE_ID }}"
>  data-page-url="{{ PAGE_URL }}"
>  data-page-title="{{ PAGE_TITLE }}">
> </div>
> ```
> 
> Then here is the script you need to include on your page to resize the iframe on each page reload.
> 
> ```
> window.addEventListener('load', function () {
>             setTimeout(() => {
>                 let scrollHeight = document.querySelector("#cusdis_id_here iframe").contentWindow.document.body.scrollHeight;
>                 document.querySelector("#cusdis_id_here iframe").style.height = scrollHeight + "px";
>             }, 3000);
>         });
> ```
> 
> I had to add the timeout because the iframe seems to take awhile to load so we want to make sure the element we're targeting actually exists in dom.
> 
> Adjust the timeout as you like, but considering comments are usually at the bottom of the page (and users usually don't scroll to the bottom straight away), it's safer to give the iframe more time to load than adjust its height prematurely.
> 
> Your mileage may vary.

Improvement to your code snippet:  
```javascript
window.addEventListener('load', function () {
    let iframe = document.querySelector("#cusdis_thread iframe");
    if (iframe) {
        let observer = new MutationObserver(() => {
            let scrollHeight = iframe.contentWindow.document.body.scrollHeight;
            iframe.style.height = scrollHeight + "px";
        });
        observer.observe(iframe.contentWindow.document.body, { childList: true, subtree: true });
    }
});
```

This uses an `observer` instead of a delay, which should also work when the comment section is lazy-loaded.

**Comment by julzhou6** [comment]
> > I was having the same problem where the height property wasn't set and I was getting a small window with a scroll bar.
> > I am using the free tier, hosted at cusdis.com and using the embed code provided.
> > I've come up with a workaround by writing a short script to resize the iframe on page load.
> > If you're using the embed code provided like I am, which looks like this
> > ```
> > <div id="cusdis_id_here"
> >  data-host="https://cusdis.com"
> >  data-page-id="{{ PAGE_ID }}"
> >  data-page-url="{{ PAGE_URL }}"
> >  data-page-title="{{ PAGE_TITLE }}">
> > </div>
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Then here is the script you need to include on your page to resize the iframe on each page reload.
> > ```
> > window.addEventListener('load', function () {
> >             setTimeout(() => {
> >                 let scrollHeight = document.querySelector("#cusdis_id_here iframe").contentWindow.document.body.scrollHeight;
> >                 document.querySelector("#cusdis_id_here iframe").style.height = scrollHeight + "px";
> >             }, 3000);
> >         });
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I had to add the timeout because the iframe seems to take awhile to load so we want to make sure the element we're targeting actually exists in dom.
> > Adjust the timeout as you like, but considering comments are usually at the bottom of the page (and users usually don't scroll to the bottom straight away), it's safer to give the iframe more time to load than adjust its height prematurely.
> > Your mileage may vary.
> 
> Improvement to your code snippet:
> 
> window.addEventListener('load', function () {
>     let iframe = document.querySelector("#cusdis_thread iframe");
>     if (iframe) {
>         let observer = new MutationObserver(() => {
>             let scrollHeight = iframe.contentWindow.document.body.scrollHeight;
>             iframe.style.height = scrollHeight + "px";
>         });
>         observer.observe(iframe.contentWindow.document.body, { childList: true, subtree: true });
>     }
> });
> This uses an `observer` instead of a delay, which should also work when the comment section is lazy-loaded.

This works perfectly, thanks!

**Comment by geoffreym** [comment]
Thanks, @4627488 and @seburbandev looks good to me. 
Hope Cusdis will natively fixe that issue

**Comment by miguelmaiquez** [comment]
I'm having the same problem. Thank you very much, [seburbandev](https://github.com/seburbandev) and [4627488](https://github.com/4627488), for the fixes. It works, although not consistenly, at least not for me. Sometimes I have to reload the page several times, or even clean the cache.

**Comment by thalskarth** [comment]
Also having same issue but couldn't solved with the scripts given

**Comment by SaySaeqo** [comment]
I have found much cleaner workaround. It is enough to just add some css for the div:
```
#cusdis_thread {
display: flex;
height: 40em; // you can change to different value
}
```

**Comment by muety** [comment]
@djyde Is this project still being worked on? 😇

**Comment by draft13** [comment]
> [@djyde](https://github.com/djyde) Is this project still being worked on? 😇

Doesn't seem to be. I've moved on from to something else. (using a discourse server as a comment backend)
